Fix-User-Guide.md

```diff
@@ -0,0 +1,252 @@
+
+```
+a : simpleField
+b : c : objectField1
+   d : objectField2
+   e : objectField3
+f : repeatedField1
+f : repeatedField2
+f : repeatedField3
+g : - listElement1
+   - listElement2
+   - listElement3
+h : - i : listObjectElement1.1
+      j : listObjectElement1.2
+    - i : listObjectElement2.1
+      j : listObjectElement2.2
+k : l : m : o : deepNestedField
+```
+
+The path for a simple string-element is adressed by stating the element name: `a`
+For the fields with deeper structure you add a dot ‘.’. The path for elements in nested objects is stated by: `b.c` or `k.l.m.o`
+
+In an data set an element sometimes an element can have multiple instances. Different data models solve this possibility differently. XML-Records can have all elements multiple times, element repition is possible and in many schemas (partly) allowed. Repeatable elements also exist in JSON and YAML but are unusual.
+
+To point to a specific element you state the index number. To adress the value `repeatedField2` the path would be `f.2`` since the repeated field is handled as list.
+Similar you adress the `listElement3` of the array/list by `g[].3`. The brackets are an array indicator created by the flux command decode-yaml. It helps to interpret an element as array also if the list has only on value.
+
+Working with nested structures and combination of arrays and objects the path is a combination of element names, dots and index numbers.
+
+`listObjectElement2.2` has the path: `h[].2.j`
+
+You do not only need the path name for your source element but also if you want to create a new element. But remember that fix as in catmandu is using repeated fields and arrays as lists so if you want to create a repeated field you have to create an array without suffic [].
+
+e.g.:
+```PERL
+copy_field("a", "z.y.x")
+```
+
+This would copy the value of z in a nested object:
+
+```
+z : 
+   y: 
+      x : simpleField
+```
+
+
+To adress paths you can use wildcards. For instance the star-wildcard: `person*` would match all simple literals with element names starting with 'person': 'person\_name', 'person\_age', etc.
+Apart from the star-wildcard, the questionmark-wildcard ('?') is supported. It matches exactly one arbitrary character.
+
+Not fully supported yet is alteration of pathes.
+
+Besides path wildcards there are array/list wildcards that are used to refrence specific elements or all elements in an array. `g[].*` adresses all strings in the array `g[]`. `g[].$append` would refrence a new element in the array at the end of the array. `g[].$last` refrences the last element in an array.
+
+## Cleaning up the transformation
+
+Since FIX is not constructing a new record stream but is manipulating the existing record you usually clean up after you transform the data. There are functions to kick out all unnecessary elements an kick out all empty elements.
```

Comment:
```suggestion
Since FIX is not constructing a new record stream but is manipulating the existing record you usually clean up after you transform the data. There are functions to remove all unnecessary elements and to remove all empty elements.
```